gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -322,6 +326,19 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
       return new AddSpecResponse<>(response);
     }
 
+    // Check quota limits against run immediately flows or adhoc flows before saving the schedule
+    if (!jobConfig.containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY) || PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false")) {
```

Comment:
I was trying to ascertain whether a scheduled+runImmediately flow that gets rejected because the quota is exceeded would still be scheduled (e.g. for the next run according to its schedule, even though running immediately is skipped).  is that so?

Reply:
The scheduled + runImmediately flow would be rejected completely here since the check is done right after compilation and before the job is added to the scheduled map.

If the flow that exists is updated to be runImmediately and exceeds quota, the runImmediate change would be rejected (and any other modifications done on the flow). The flow would retain its previous state and then ran on its old schedule.

If a flow that doesn't exist has runImmediately and exceeds quota, it is rejected from the system entirely.

So the current model in the PR is that any changes that involve running a flow immediately when the quota is exceeded is rejected. I think it's possible to move to a future state where changes to a flow is not dependent on the flow's execution schedule, but it would require a lot of refactoring. I think it can be grouped with other changes we talked about, such as the faster feedback when running adhoc flows. I think a possible future state would be where users can save flows independently of whether or not the flow is scheduled, and the execution schedule would be treated entirely separate (similar to ADF and some other systems).

Reply:
it feels potentially confusing that the flow would not be scheduled at all (for the future) just because the quota is exceeded now.  wouldn't it be safe to `scheduleJob` anyway (even if we don't believe the first, runImmediately execution would succeed)?